packages/bpk-react-utils/src/BpkDialogWrapper/BpkDialogWrapper.tsx

```diff
@@ -0,0 +1,181 @@
+ * limitations under the License.
+ */
+import type { ReactNode } from "react";
+import { useEffect, useRef, useState } from "react";
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import CSSTransition from 'react-transition-group/CSSTransition';
+
+import cssModules from '../cssModules';
+
+import STYLES from './BpkDialogWrapper.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export interface Props {
+  ariaLabelledby: string;
+  children: ReactNode;
+  closeOnEscPressed?: boolean;
+  closeOnScrimClick?: boolean;
+  dialogClassName?: string;
+  id: string | undefined;
+  isOpen: boolean;
+  onClose: (
+    arg0?: Event,
+    arg1?: {
+      source: 'ESCAPE' | 'DOCUMENT_CLICK';
+    },
+) => void | null;
+  exiting?: boolean;
+  transitionClassNames?: {
+    appear?: string,
+    appearActive?: string,
+    exit?: string
+  };
+  timeout?: {appear?: number, exit?: number};
+};
+
+type DialogProps = {
+  isDialogOpen: boolean;
+};
+
+// TODO: this check if the browser support the HTML dialog element. We can remove it once we drop support as a business for Safari 14
+const dialogSupported = typeof HTMLDialogElement === 'function';
+
+const setPageProperties = ({ isDialogOpen }: DialogProps) => {
+  document.body.style.overflowY = isDialogOpen ? 'hidden' : 'visible';
+
+  if (!dialogSupported) {
+    document.body.style.position = isDialogOpen ? 'fixed' : 'relative';
+    document.body.style.width = isDialogOpen ? '100%' : 'auto';
+  }
+};
+
+export const BpkDialogWrapper = (props: Props) => {
+  const {
+    ariaLabelledby,
+    children,
+    closeOnEscPressed = false,
+    closeOnScrimClick = false,
+    dialogClassName = '',
+    exiting = false,
+    id,
+    isOpen,
+    onClose,
+    timeout = {appear: 0, exit: 0},
+    transitionClassNames = {}
+  } = props;
+
+  const ref = useRef<HTMLDialogElement>(null);
+  const [dialogTarget, setDialogTarget] = useState<HTMLElement | null>(null);
+
+  useEffect(() => {
+    const dialog = document.getElementById(`${id}`);
+    const dialogWithPolyfill = document.getElementById(`${id}-polyfill`);
+
+    const handleBackdropClick = (modal: HTMLElement | null) => {
+      if (closeOnScrimClick && modal) {
+        modal.addEventListener('click', (event: MouseEvent | PointerEvent) => {
+          const { target } = event;
+
+          if (target === modal) {
+            onClose(event, {source: "DOCUMENT_CLICK"});
+            event.stopPropagation();
+          }
+        });
+      }
+    };
+
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (closeOnEscPressed
+        && event.key === 'Escape'
+        && (!dialogWithPolyfill || event.target === dialogWithPolyfill)) {
+        onClose(event, {source: "ESCAPE"});
+      }
+      event.stopPropagation();
+    };
+
+    if (isOpen) {
+      // There is a bug on older versions of browser using chromium (chrome, firefox, edge >114) where the dialog got an open attribute even before it is opened.
```

Comment:
Do we know which versions have this bug and are they still in the range of our supported browser list?

Reply:
Im not sure what backpack's supported browser list is, but here is a list of versions that is supported https://caniuse.com/mdn-api_htmldialogelement_open